### PR TITLE
[WIP] Introduce projection expression syntax

### PR DESCRIPTION
**Pull request by birkskyum**
The ability to use adaptive transitions isn't captured well by the style spec yet. It could very well be an expression for world projections

see https://github.com/maplibre/maplibre-style-spec/pull/888#issuecomment-2461318216

## Launch Checklist

 - [X] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [ ] Briefly describe the changes in this PR.
 - [ ] Link to related issues.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

**Comment by HarelM** [comment]
I would consider waiting for the spec change and then add the spec package change as part of this PR and change whatever is needed to support the new spec, what do you think?

**Comment by birkskyum** [comment]
Sounds good!

**Comment by HarelM** [comment]
Another approach you might want to try out is using `npm link`, change the code in the style-spec package, use that here, make sure you have everything you need from the spec and then update the spec PR, we will merge it, release a new version and then continue with this PR.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 ### 🐞 Bug fixes
 - Fix `line` layer on Globe with Terrain Elevation
 - ⚠️ Fix level of detail at high pitch angle by changing which tiles to load ([#3983](https://github.com/maplibre/maplibre-gl-js/issues/3983))
+- Make `globe` projection non-adaptive, until spec catches up
```
[the remark this answers is not in the document]
Comment (birkskyum) [reply CHANGELOG.md:9]: @kubapelc , adding support for the non-adaptive globe here:
- #5023

**Pull request closed without merging**